zeapp/server/src/main/kotlin/de/berlindroid/zekompanion/server/Main.kt

```diff
@@ -0,0 +1,103 @@
+@file:JvmName("Main")
+
+package de.berlindroid.zekompanion.server
+
+import de.berlindroid.zekompanion.server.routers.imageBin
+import de.berlindroid.zekompanion.server.routers.imagePng
+import de.berlindroid.zekompanion.server.routers.index
+import io.ktor.serialization.kotlinx.json.*
+import io.ktor.server.application.*
+import io.ktor.server.engine.*
+import io.ktor.server.http.content.*
+import io.ktor.server.plugins.contentnegotiation.*
+import io.ktor.server.routing.*
+import io.ktor.server.tomcat.Tomcat
+import java.io.File
+import java.security.KeyStore
+
+private const val LOCAL_HTTP_PORT = 8000
+private const val LOCAL_TLS_PORT = 8443
+private const val SSL_PASSWORD_ENV = "SSL_CERTIFICATE_PASSWORD"
+private const val KEYSTORE_RESOURCE_FILE = "/tmp/keystore.jks"
+
+
+fun main(args: Array<String>) {
+    val keyPassword = try {
+        System.getenv(SSL_PASSWORD_ENV)
+    } catch (e: Exception) {
+        null
+    }
+
+    val keyStore: KeyStore? = loadKeyStore(keyPassword)
+    val serverPort = extractServerPort(args, keyStore)
+    println("Serving on port $serverPort.")
+
+    embeddedServer(
+        Tomcat,
+        environment = applicationEngineEnvironment {
+            injectTLSIfNeeded(keyStore, keyPassword)
+
+            module {
+                install(ContentNegotiation) {
+                    json()
+                }
+
+                routing {
+                    staticResources("/", "static") {
```

Comment:
Something to check - is `.git` and actually everything `\..*` excluded by default?

Reply:
That is the magic of `staticResources` vs `staticFiles` ([Link to Docs](https://ktor.io/docs/serving-static-content.html#resources)): Our solution only serves the [resources](../tree/main/zeapp/server/src/main/resources/static), i.e. only the [repository index.html](../tree/main/zeapp/server/src/main/resources/static/index.html) is available online: [index](zebadge.app/index.html) (same for the js).

But an non existing file will return [a 404](https://http.cat/404) and an empty page. Also relative path obviously don't work.

I feel safe here.